packages/dev/core/src/Engines/abstractEngine.ts

```diff
@@ -2655,4 +2657,64 @@ export abstract class AbstractEngine {
      * By default, this will create a Database object if the workload has been embedded.
      */
     public static OfflineProviderFactory: (urlToScene: string, callbackManifestChecked: (checked: boolean) => any, disableManifestCheck: boolean) => IOfflineProvider;
+
+    /**
+     * Will flag all materials in all scenes in all engines as dirty to trigger new shader compilation
+     * @param flag defines which part of the materials must be marked as dirty
+     * @param predicate defines a predicate used to filter which materials should be affected
+     */
+    public static MarkAllMaterialsAsDirty(flag: number, predicate?: (mat: Material) => boolean): void {

```

Comment:
Is is not possible to keep this out of abstract/thin ?

Reply:
agreed on that one it may mpact thinengine size

Reply:
I actually forgot about this and didn't change it. Please check the most recent commit

In conclusion, we should not make child classes implement MarkAllMaterialsAsDirty, because the implementation of Material is not engine specific

Reply:
Agreed and this is very small. I think it is ok to keep it here